go/cmd/vtctldclient/command/shards.go

```diff
@@ -52,6 +53,38 @@ that shard.`,
 		Args:                  cobra.MinimumNArgs(1),
 		RunE:                  commandDeleteShards,
 	}
+	// GenerateShardRanges outputs a set of shard ranges assuming a (mostly)
+	// equal distribution of N shards.
+	GenerateShardRanges = &cobra.Command{
+		Use:                   "GenerateShardRanges <num_shards>",
+		Short:                 "Print a set of shard ranges assuming a keyspace with N shards.",
+		DisableFlagsInUseLine: true,
+		Args:                  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			n, err := strconv.ParseInt(cmd.Flags().Arg(0), 10, 64)
+			if err != nil {
+				return err
+			}
+
+			cli.FinishedParsing(cmd)
+
+			shards, err := key.GenerateShardRanges(int(n))
+			if err != nil {
+				return err
+			}
+
+			data, err := cli.MarshalJSON(shards)
+			if err != nil {
+				return err
+			}
+
+			fmt.Printf("%s\n", data)
+			return nil
+		},
+		Annotations: map[string]string{
+			skipClientCreationKey: "true",
+		},
```

Comment:
It doesn't make sense to use `VtctldClientProtocol` for this and set it to `nil` or `"none"` for this? At first glance it feels more intuitive to me given the existing `grpc` and `local` values.

Reply:
are you saying to do something like:

```go
var GenerateShardRanges = &cobra.Command{
	Annotations: map[string]string{
		"protocol": "local",
	}
}
```

?

If you're saying to avoid using `Annotations` altogether in favor of "just" checking the `VtctldClientProtocol` variable ... there's no way to force that to a specific value for _just_ this command, as the root pre-run (where that `ensureServerArg/getClient` func is called) runs before any funcs on a child command such as this one.

Reply:
I thought we might be able to set `cmd.VtctldClientProtocol="none"` here. If not (and I think you're saying NOT here), then never mind. Thanks!